rubocop-klaxit/cops/no_undefined_variable_usage.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    class NoUndefinedVariableUsage < Cop
+      MSG = "Variable used before being defined"
+
+      def_node_matcher :has_param?, <<~PATTERN
+        ({def defs block} <(args <(_ %1 ...)...>) ...>)
+      PATTERN
+
+      def_node_matcher :masgn_with_sym, <<~PATTERN
+        (masgn (mlhs (lvasgn $_)*) ...)
+      PATTERN
+
+      def_node_search :lvar_with_sym, "(lvar %1)"
+
+      # :nodoc:
+      def on_lvasgn(node)
+        sym = node.children[0]
+
+        return if lvar_with_sym(node, sym).all? { defined_before?(_1, sym) }
+
+        add_offense(node)
+      end
+
+      # :nodoc:
+      def on_masgn(node)
```

Comment:
```suggestion
      # Multiple vars assignment
      # @example `foo, bar = 1, 2`
      #   (masgn
      #     (mlhs
      #       (lvasgn :foo)
      #       (lvasgn :bar))
      #     (array
      #       (int 1)
      #       (int 2)))

      def on_masgn(node)
```

Also, since `masgn` contains `lvasgn`, is there really a purpose of having that method? I feel like it is a bit unnecessary